src/tests/dune

```diff
@@ -29,8 +29,5 @@
    (pps
       ppx_deriving.show
    ))
+ (deps (source_tree ../tests) (source_tree ../cli/tests/e2e/snapshots))
```

Comment:
This would require a big comment like 
; If you add test code that depends on some data, you need to update the deps below otherwise
; dune buld will just copy the data under _build/ and will not use the latest version of this data, even if you modify
; this data.

Reply:
done in 6f6946960d23f7a255e32d0fcb0018696d4b5925 -- is this understandable?

With this PR, I can "dune runtest" (or make core-test), then modify any rule, and a subsequent make core-test will rerun the test suite and report failure (if I messed up the rule).

So, the "will not use the latest version of this data, even if you modify this data." is not true anymore, since now dune knows that if ../tests (or ../cli/tests/e2e/snapshots) is modified, there's need to re-run the test suite.